tutorials/ann_hnswlib.ipynb

```diff
@@ -0,0 +1,394 @@
+      "Total items = 4926\n",
+      "\n",
+      "[MF] Training started!\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "8136f6f5bca44cde89e682ab2a62dd24",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "  0%|          | 0/25 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Optimization finished!\n",
+      "\n",
+      "[MF] Evaluation started!\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "d917bf14c68548a490d8b9f38562320b",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Ranking:   0%|          | 0/8233 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "TEST:\n",
+      "...\n",
+      "   |    AUC | Recall@20 | Train (s) | Test (s)\n",
+      "-- + ------ + --------- + --------- + --------\n",
+      "MF | 0.8517 |    0.0440 |    1.1022 |  11.8385\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "data = netflix.load_feedback(variant=\"small\", reader=Reader(bin_threshold=1.0))\n",
+    "\n",
+    "ratio_split = RatioSplit(\n",
+    "    data=data,\n",
+    "    test_size=0.1,\n",
+    "    rating_threshold=1.0,\n",
+    "    exclude_unknowns=True,\n",
+    "    verbose=True,\n",
+    ")\n",
+    "\n",
+    "mf = MF(\n",
+    "    k=50, \n",
+    "    max_iter=25, \n",
+    "    learning_rate=0.01, \n",
+    "    lambda_reg=0.02, \n",
+    "    use_bias=False,\n",
+    "    verbose=True,\n",
+    "    seed=123,\n",
+    ")\n",
+    "\n",
+    "auc = cornac.metrics.AUC()\n",
+    "rec_20 = cornac.metrics.Recall(k=20)\n",
+    "\n",
+    "cornac.Experiment(\n",
+    "    eval_method=ratio_split,\n",
+    "    models=[mf],\n",
+    "    metrics=[auc, rec_20],\n",
+    "    user_based=True,\n",
+    ").run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b00b5236-1a94-43a9-bfb9-6dc4623bb33c",
+   "metadata": {},
+   "source": [
+    "## Building index for ANN recommender\n",
+    "\n",
+    "After MF model is trained, we need to wrap it with an ANN recommender. We employ Cornac built-in HNSWLibANN which implements [HNSW algorithm](https://arxiv.org/abs/1603.09320) for building index and doing approximate K-nearest neighbor search. More on how to tune the hyper-parameters at https://github.com/nmslib/hnswlib."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "fa280b9d-ec04-41eb-9de2-acfb67fbeb80",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ann = HNSWLibANN(\n",
+    "    recom=mf,\n",
+    "    M=16,\n",
+    "    ef_construction=100,\n",
+    "    ef=50,\n",
```

Comment:
Can we add a comment to stress that this parameter must be greater than K for a top-K recommendation list

Reply:
added